// @vkontakte/vkjs/lib/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "animate", {
  enumerable: true,
  get: function get() {
    return _animate.animate;
  }
});
Object.defineProperty(exports, "animationEvent", {
  enumerable: true,
  get: function get() {
    return _animate.animationEvent;
  }
});
Object.defineProperty(exports, "transitionEvent", {
  enumerable: true,
  get: function get() {
    return _animate.transitionEvent;
  }
});
Object.defineProperty(exports, "waitAnimationEnd", {
  enumerable: true,
  get: function get() {
    return _animate.waitAnimationEnd;
  }
});
Object.defineProperty(exports, "cancelWaitAnimationEnd", {
  enumerable: true,
  get: function get() {
    return _animate.cancelWaitAnimationEnd;
  }
});
Object.defineProperty(exports, "waitTransitionEnd", {
  enumerable: true,
  get: function get() {
    return _animate.waitTransitionEnd;
  }
});
Object.defineProperty(exports, "cancelWaitTransitionEnd", {
  enumerable: true,
  get: function get() {
    return _animate.cancelWaitTransitionEnd;
  }
});
Object.defineProperty(exports, "classNames", {
  enumerable: true,
  get: function get() {
    return _classNames.classNames;
  }
});
Object.defineProperty(exports, "copyTextToClipboard", {
  enumerable: true,
  get: function get() {
    return _clipboard.copyTextToClipboard;
  }
});
Object.defineProperty(exports, "isNumeric", {
  enumerable: true,
  get: function get() {
    return _common.isNumeric;
  }
});
Object.defineProperty(exports, "isPassiveEventsSupported", {
  enumerable: true,
  get: function get() {
    return _detections.isPassiveEventsSupported;
  }
});
Object.defineProperty(exports, "isSmoothScrollSupported", {
  enumerable: true,
  get: function get() {
    return _detections.isSmoothScrollSupported;
  }
});
Object.defineProperty(exports, "isEqual", {
  enumerable: true,
  get: function get() {
    return _equal.isEqual;
  }
});
Object.defineProperty(exports, "sumArray", {
  enumerable: true,
  get: function get() {
    return _arrays.sumArray;
  }
});
Object.defineProperty(exports, "averageArray", {
  enumerable: true,
  get: function get() {
    return _arrays.averageArray;
  }
});
Object.defineProperty(exports, "uniqueArray", {
  enumerable: true,
  get: function get() {
    return _arrays.uniqueArray;
  }
});
Object.defineProperty(exports, "shuffleArray", {
  enumerable: true,
  get: function get() {
    return _arrays.shuffleArray;
  }
});
Object.defineProperty(exports, "chunkArray", {
  enumerable: true,
  get: function get() {
    return _arrays.chunkArray;
  }
});
Object.defineProperty(exports, "omitFromArray", {
  enumerable: true,
  get: function get() {
    return _arrays.omitFromArray;
  }
});
Object.defineProperty(exports, "difference", {
  enumerable: true,
  get: function get() {
    return _arrays.difference;
  }
});
Object.defineProperty(exports, "asyncImportLoader", {
  enumerable: true,
  get: function get() {
    return _async.asyncImportLoader;
  }
});
Object.defineProperty(exports, "getCookie", {
  enumerable: true,
  get: function get() {
    return _cookie.getCookie;
  }
});
Object.defineProperty(exports, "SECONDS_IN_THE_DAY", {
  enumerable: true,
  get: function get() {
    return _date.SECONDS_IN_THE_DAY;
  }
});
Object.defineProperty(exports, "isDateToday", {
  enumerable: true,
  get: function get() {
    return _date.isDateToday;
  }
});
Object.defineProperty(exports, "isDateYesterday", {
  enumerable: true,
  get: function get() {
    return _date.isDateYesterday;
  }
});
Object.defineProperty(exports, "isDateTomorrow", {
  enumerable: true,
  get: function get() {
    return _date.isDateTomorrow;
  }
});
Object.defineProperty(exports, "isSameDate", {
  enumerable: true,
  get: function get() {
    return _date.isSameDate;
  }
});
Object.defineProperty(exports, "getBeginningOfDay", {
  enumerable: true,
  get: function get() {
    return _date.getBeginningOfDay;
  }
});
Object.defineProperty(exports, "isLeapYear", {
  enumerable: true,
  get: function get() {
    return _date.isLeapYear;
  }
});
Object.defineProperty(exports, "getLastDayOfMonth", {
  enumerable: true,
  get: function get() {
    return _date.getLastDayOfMonth;
  }
});
Object.defineProperty(exports, "getStartOfWeek", {
  enumerable: true,
  get: function get() {
    return _date.getStartOfWeek;
  }
});
Object.defineProperty(exports, "addDays", {
  enumerable: true,
  get: function get() {
    return _date.addDays;
  }
});
Object.defineProperty(exports, "createDateFromUnixTimestamp", {
  enumerable: true,
  get: function get() {
    return _date.createDateFromUnixTimestamp;
  }
});
Object.defineProperty(exports, "getUnixTimestampFromDate", {
  enumerable: true,
  get: function get() {
    return _date.getUnixTimestampFromDate;
  }
});
Object.defineProperty(exports, "convertDateToInputFormat", {
  enumerable: true,
  get: function get() {
    return _date.convertDateToInputFormat;
  }
});
Object.defineProperty(exports, "formatDuration", {
  enumerable: true,
  get: function get() {
    return _duration.formatDuration;
  }
});
Object.defineProperty(exports, "noop", {
  enumerable: true,
  get: function get() {
    return _functions.noop;
  }
});
Object.defineProperty(exports, "throttle", {
  enumerable: true,
  get: function get() {
    return _functions.throttle;
  }
});
Object.defineProperty(exports, "debounce", {
  enumerable: true,
  get: function get() {
    return _functions.debounce;
  }
});
Object.defineProperty(exports, "getOffsetRect", {
  enumerable: true,
  get: function get() {
    return _getOffsetRect.getOffsetRect;
  }
});
Object.defineProperty(exports, "leadingZero", {
  enumerable: true,
  get: function get() {
    return _numbers.leadingZero;
  }
});
Object.defineProperty(exports, "formatNumber", {
  enumerable: true,
  get: function get() {
    return _numbers.formatNumber;
  }
});
Object.defineProperty(exports, "deleteObjectKeys", {
  enumerable: true,
  get: function get() {
    return _objects.deleteObjectKeys;
  }
});
Object.defineProperty(exports, "querystring", {
  enumerable: true,
  get: function get() {
    return _querystring.querystring;
  }
});
Object.defineProperty(exports, "getRandomInt", {
  enumerable: true,
  get: function get() {
    return _random.getRandomInt;
  }
});
Object.defineProperty(exports, "getRandomString", {
  enumerable: true,
  get: function get() {
    return _random.getRandomString;
  }
});
Object.defineProperty(exports, "hasReactNode", {
  enumerable: true,
  get: function get() {
    return _react_utils.hasReactNode;
  }
});
Object.defineProperty(exports, "isPrimitiveReactNode", {
  enumerable: true,
  get: function get() {
    return _react_utils.isPrimitiveReactNode;
  }
});
Object.defineProperty(exports, "isObjectLike", {
  enumerable: true,
  get: function get() {
    return _type_checkers.isObjectLike;
  }
});
Object.defineProperty(exports, "isArray", {
  enumerable: true,
  get: function get() {
    return _type_checkers.isArray;
  }
});
Object.defineProperty(exports, "isObject", {
  enumerable: true,
  get: function get() {
    return _type_checkers.isObject;
  }
});
Object.defineProperty(exports, "isUndefined", {
  enumerable: true,
  get: function get() {
    return _type_checkers.isUndefined;
  }
});
Object.defineProperty(exports, "isFunction", {
  enumerable: true,
  get: function get() {
    return _type_checkers.isFunction;
  }
});
Object.defineProperty(exports, "isFormData", {
  enumerable: true,
  get: function get() {
    return _type_checkers.isFormData;
  }
});
Object.defineProperty(exports, "isString", {
  enumerable: true,
  get: function get() {
    return _type_checkers.isString;
  }
});
Object.defineProperty(exports, "isNumber", {
  enumerable: true,
  get: function get() {
    return _type_checkers.isNumber;
  }
});
Object.defineProperty(exports, "escapeRegExp", {
  enumerable: true,
  get: function get() {
    return _regexp.escapeRegExp;
  }
});
Object.defineProperty(exports, "localStorage", {
  enumerable: true,
  get: function get() {
    return _storage.localStorage;
  }
});
Object.defineProperty(exports, "sessionStorage", {
  enumerable: true,
  get: function get() {
    return _storage.sessionStorage;
  }
});
Object.defineProperty(exports, "hasMouse", {
  enumerable: true,
  get: function get() {
    return _InputUtils.hasMouse;
  }
});
Object.defineProperty(exports, "hasHover", {
  enumerable: true,
  get: function get() {
    return _InputUtils.hasHover;
  }
});
Object.defineProperty(exports, "hasTouchEvents", {
  enumerable: true,
  get: function get() {
    return _InputUtils.hasTouchEvents;
  }
});
Object.defineProperty(exports, "hasTouch", {
  enumerable: true,
  get: function get() {
    return _InputUtils.hasTouch;
  }
});
Object.defineProperty(exports, "canUseDOM", {
  enumerable: true,
  get: function get() {
    return _dom.canUseDOM;
  }
});
Object.defineProperty(exports, "canUseEventListeners", {
  enumerable: true,
  get: function get() {
    return _dom.canUseEventListeners;
  }
});
Object.defineProperty(exports, "onDOMLoaded", {
  enumerable: true,
  get: function get() {
    return _dom.onDOMLoaded;
  }
});
Object.defineProperty(exports, "IPHONE_SAFARI_BOTTOM_BAR", {
  enumerable: true,
  get: function get() {
    return _IOSDetections.IPHONE_SAFARI_BOTTOM_BAR;
  }
});
Object.defineProperty(exports, "IPHONE_X_SAFARI_BOTTOM_BAR", {
  enumerable: true,
  get: function get() {
    return _IOSDetections.IPHONE_X_SAFARI_BOTTOM_BAR;
  }
});
Object.defineProperty(exports, "IPHONE_KEYBOARD_REJECT_OFFSET", {
  enumerable: true,
  get: function get() {
    return _IOSDetections.IPHONE_KEYBOARD_REJECT_OFFSET;
  }
});
Object.defineProperty(exports, "IOS_NO_KEYBOARD_ALLOWED_OFFSET", {
  enumerable: true,
  get: function get() {
    return _IOSDetections.IOS_NO_KEYBOARD_ALLOWED_OFFSET;
  }
});
Object.defineProperty(exports, "detectIOS", {
  enumerable: true,
  get: function get() {
    return _IOSDetections.detectIOS;
  }
});
Object.defineProperty(exports, "isIPad", {
  enumerable: true,
  get: function get() {
    return _IOSDetections.isIPad;
  }
});
Object.defineProperty(exports, "isIPhone", {
  enumerable: true,
  get: function get() {
    return _IOSDetections.isIPhone;
  }
});
Object.defineProperty(exports, "isIOS", {
  enumerable: true,
  get: function get() {
    return _IOSDetections.isIOS;
  }
});
Object.defineProperty(exports, "isIPadOS", {
  enumerable: true,
  get: function get() {
    return _IOSDetections.isIPadOS;
  }
});
Object.defineProperty(exports, "iosMajor", {
  enumerable: true,
  get: function get() {
    return _IOSDetections.iosMajor;
  }
});
Object.defineProperty(exports, "iosMinor", {
  enumerable: true,
  get: function get() {
    return _IOSDetections.iosMinor;
  }
});
Object.defineProperty(exports, "isWKWebView", {
  enumerable: true,
  get: function get() {
    return _IOSDetections.isWKWebView;
  }
});
Object.defineProperty(exports, "isScrollBasedViewport", {
  enumerable: true,
  get: function get() {
    return _IOSDetections.isScrollBasedViewport;
  }
});
Object.defineProperty(exports, "isIPhoneX", {
  enumerable: true,
  get: function get() {
    return _IOSDetections.isIPhoneX;
  }
});
Object.defineProperty(exports, "isIOSChrome", {
  enumerable: true,
  get: function get() {
    return _IOSDetections.isIOSChrome;
  }
});
Object.defineProperty(exports, "isLandscapePhone", {
  enumerable: true,
  get: function get() {
    return _IOSDetections.isLandscapePhone;
  }
});
Object.defineProperty(exports, "checkIPadOS", {
  enumerable: true,
  get: function get() {
    return _IOSDetections.checkIPadOS;
  }
});

var _animate = require("./animate");

var _classNames = require("./classNames");

var _clipboard = require("./clipboard");

var _common = require("./common");

var _detections = require("./detections");

var _equal = require("./equal");

var _arrays = require("./arrays");

var _async = require("./async");

var _cookie = require("./cookie");

var _date = require("./date");

var _duration = require("./duration");

var _functions = require("./functions");

var _getOffsetRect = require("./getOffsetRect");

var _numbers = require("./numbers");

var _objects = require("./objects");

var _querystring = require("./querystring");

var _random = require("./random");

var _react_utils = require("./react_utils");

var _type_checkers = require("./type_checkers");

var _regexp = require("./regexp");

var _storage = require("./storage");

var _InputUtils = require("./InputUtils");

var _dom = require("./dom");

var _IOSDetections = require("./IOSDetections");
